alyssa_mckee/Django/belt_reviwer/apps/book_app/templates/book_app/book.html

```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE HTML>
+<html lang="en-US">
+<head>
+	<meta charset="UTF-8">
+	<title>{{book.title}}</title>
+	
+</head>
+<body>
+	<header>
+		<a href="{% url 'dashboard' %}">Home</a>
+		<a href="{% url 'logout' %}">Logout</a>
+	
+	</header>
+	
+	<h1>{{book.title}}</h1>
+	<h3>Author: {{book.author.name}}</h3>
+	
+	<h2>Reviews:</h2>
+	
+	
+	{% for review in reviews %}
+			<div class="review">
+			<div class="rating">Rating: 
+			
+			{{review.rating}}
+			{% load static %}
+			{% if review.rating > 0 %}
```

Comment:
I bet there is a way this logic could be simplified with a loop of some kind.

Reply:
I tried that first, but django templates don't support `for x in range(y)` specifically the `range(y)` part. that's the only way i could think of to get around it at the time.

Reply:
I still can't think of any good way to do it.

if i were passing just one rating to the page, then i could just convert it to a range in the views and pass it in to the context. but i'm not, i'm sending the whole book, which has multiple reviews with ratings. I'd have to separate out the ratings, convert them to ranges, then somehow put them back with their correct reviews in the template.

 i could store the rating as a different type like maybe a string with length of the rating (like "FFFEE" where F is a full star, E is empty star, and perhaps support H for half star ) that way I could iterate through with a `for in` and load a full star for every 'F' (i think)
(of course then if i ever just wanted the number value of the rating i'd have to convert it back) 

or by writing some embedded javascript in the head to handle the actual looping part, like 
`<div class="loop" times="{{ review.rating }}">{% static 'star image' %}</div>`  in the body
and a script in the head that just selects all the .loop class and repeats the inner html equal to the times attr

or by writing my own template filter, which i still have to go look into how to do.

Reply:
Yeah, those are all definitely good solutions. I would suggest looking into custom django template tags. https://docs.djangoproject.com/en/1.11/howto/custom-template-tags/